packages/react/src/components/ListBox/ListBoxMenuItem.tsx

```diff
@@ -74,18 +80,27 @@ export type ListBoxMenuItemComponent = ForwardRefReturn<
  */
 const ListBoxMenuItem = React.forwardRef<HTMLLIElement, ListBoxMenuItemProps>(
   function ListBoxMenuItem(
-    {
-      children,
-      isActive = false,
-      isHighlighted = false,
-      title,
-      ...rest
-    }: ListBoxMenuItemProps,
+    { children, isActive = false, isHighlighted = false, title, ...rest },
     forwardedRef: ListBoxMenuItemForwardedRef
   ) {
     const prefix = usePrefix();
-    const ref = useRef(null);
-    const isTruncated = useIsTruncated(forwardedRef?.menuItemOptionRef || ref);
+    const localRef = useRef<HTMLDivElement>(null);
+
+    // Try to get a RefObject from `forwardedRef.menuItemOptionRef`.
+    // If it's a callback ref (or not provided), fall back to `localRef`.
```

Comment:
Would it make sense to test this with a mocked callback ref function?

https://github.com/carbon-design-system/carbon/blob/7b9a05be22b9bee34f70299f7568f1aeb6e7c06a/packages/icon-build-helpers/src/builders/react/components/__tests__/Icon-test.js#L40-L47